### DISCUSSION TITLE
Chronics input data

### DISCUSSION THREAD

**Discussion by rajarshiroychowdhury** [rte-france/Grid2Op#268] [Q&A] [answered]
Hi,

Just a quick question! When playing with the L2RPN Baseline agents (let's say the example with DuelDQN), do I need to specify the path of the downloaded grid data somewhere? I think even when I am creating the environment with Test = False, the summary still shows results for just two chronics (000 and 001)? Am sure am missing something obvious here! Thanks for the help!

**Comment by BDonnot** [comment, marked as answer]
Hello,

You should be able to use the correct environment just by calling (for neurips track1 for example):
```python
import grid2op
env = grid2op.make("l2rpn_neurips_2020_track1_small")
```
For the environment of track 1 for example.

A full list of the available environments is accessible with
```python
import grid2op
grid2op.list_available_remote_env()
```

As said in the warning, do not use `test=True`  for real agent training.

Benjamin

**Reply by rajarshiroychowdhury** [reply]
Hi Benjamin,

Thanks for your reply! Maybe I should have been more clear! Everything that you have said, I am sort of aware of. However, even if I create the environment with let's say
env_name = "rte_case14_redisp"
env = make(env_name)

and then let's say, for example, I follow the 04_TrainingAnAgent.ipynb notebook, my question is, why does the training after the summary, shows results for just two chronics? I just see there are 2 folders, each corresponding to a chronics. Shouldn't I have results for all the downloaded chronics for the above environment? Or maybe am missing something here? Thanks for your continued help!

Regards,
Raj

**Reply by BDonnot** [reply]
Hello,

I don't recommend to use this environment.

Real, properly calibrated environment are listed with the command `list_available_remote_env()`

You can use the `l2rpn_case14_sandbox` environment instead of the one you're trying to use, which counts indeed only a few data

More information here https://grid2op.readthedocs.io/en/latest/available_envs.html#description-of-some-environments

**Reply by rajarshiroychowdhury** [reply]
Hi Benjamin,

I tried with the environment you suggested 
env_name = "l2rpn_case14_sandbox"  
env = grid2op.make(env_name), 

but I think it's actually showing me the same results. Like after all the training steps, when I write
 
import os
os.listdir(path_save_results)

The output I see is 

['0000',
 '0001',
 'dict_action_space.json',
 'dict_attack_space.json',
 'dict_env_modification_space.json',
 'dict_observation_space.json']

I guess it's again just taking the 0000 and 0001 chronics and not using the other downloaded chronics data! 

There should be a way to ask the  DDQN agent to train on all the chronics, isn't it? When I check the default grid2op_data folder where the data gets downloaded, I see there are many other chronics folder after 0000 and 0001 but while training it always takes the first two and that's it!  What am I missing?

**Reply by BDonnot** [reply]
Hello, can I see the code you use to generate these logs ?
I think you are either selecting only two chronics as test (and test on these ones) or you are evalutating your agent (once trained) on only two scenarios (hence the only two scenarios that you see here)

**Reply by rajarshiroychowdhury** [reply]
Here you go! Thanks much for your time!
[RL_Project_Power_System_Agent.zip](https://github.com/rte-france/Grid2Op/files/7623802/RL_Project_Power_System_Agent.zip)

**Reply by BDonnot** [reply]
That is what i thought.

In:


```python
evaluated_agent, res_runner = evaluate(env,
                                       name=agent_name,
                                       load_path=save_path,
                                       logs_path=path_save_results,
                                       nb_episode=2,
                                       nb_process=1,
                                       max_steps=max_eval_step,
                                       verbose=True,
                                       save_gif=False)
```

You ask to evaluate your agent on 2 episode (`nb_episode=2`) so it's perfectly consistent that in the result folder, you will see exactly 2 episodes

**Reply by rajarshiroychowdhury** [reply]
I see! That was so dumb on my part! Thanks much! Appreciate this very much!

**Reply by BDonnot** [reply]
No problem, happy to haved helped :-)

Also don't forget that the chronics you see there are the chronics you test on, and NOT the chronics you trained on.
